toolkit/themes/shared/design-system/config.js

```diff
@@ -1,11 +1,101 @@
 
 /* eslint-env node */
 
 const StyleDictionary = require("style-dictionary");
+const { formattedVariables } = StyleDictionary.formatHelpers;
+
+const MEDIA_QUERY_PROPERTY_MAP = {
+  "forced-colors": "forcedColors",
+  "prefers-contrast": "prefersContrast",
+};
+
+/**
+ * Formats built CSS to include "prefers-contrast" and "forced-colors" media
+ * queries.
+ *
+ * @param {object} args
+ *  Formatter arguments provided by style-dictionary. See more at
+ *  https://amzn.github.io/style-dictionary/#/formats?id=formatter
+ * @returns {string}
+ *  Formatted CSS including media queries.
+ */
+function hcmFormatter(args) {
+  return (
+    formatTokens({ args }) +
+    formatTokens({ mediaQuery: "prefers-contrast", args }) +
+    formatTokens({ mediaQuery: "forced-colors", args })
+  );
+}
+
+/**
+ * Formats a subset of tokens into CSS. Wraps token CSS in a media query when
+ * applicable.
+ *
+ * @param {object} tokenArgs
+ * @param {string} [tokenArgs.mediaQuery]
+ *  Media query formatted CSS should be wrapped in. This is used
+ *  to determine what property we are parsing from the token values.
+ * @param {object} tokenArgs.args
+ *  Formatter arguments provided by style-dictionary. See more at
+ *  https://amzn.github.io/style-dictionary/#/formats?id=formatter
+ * @returns {string} Tokens formatted into a CSS string.
+ */
+function formatTokens({ mediaQuery, args }) {
+  let prop = MEDIA_QUERY_PROPERTY_MAP[mediaQuery] ?? "value";
+  let dictionary = Object.assign({}, args.dictionary);
+  let tokens = [];
+
+  dictionary.allTokens.forEach(token => {
+    let value = token[prop] || token.original.value[prop]
+    if (value && typeof value !== "object") {
+      let formattedToken = transformTokenValue(token, prop, dictionary);
+      tokens.push(formattedToken);
+    }
+  });
+
+  dictionary.allTokens = dictionary.allProperties = tokens;
+
+  let formattedVars = formattedVariables({
+    format: "css",
+    dictionary,
+    outputReferences: args.options.outputReferences,
+  });
+
+  if (mediaQuery) {
+    return `
+@media (${mediaQuery}) {
+  :root {
+  ${formattedVars.split("\n").join(`\n  `)}
+  }
+}
+`;
+  }
+
+  return `:root {\n${formattedVars}\n}\n`;
+}
+
+/**
+ * Takes a token object and changes "value" based on the supplied prop. Also
+ * preserves variable references when necessary.
+ *
+ * @param {object} token - Token object parsed from JSON by style-dictionary.
+ * @param {string} prop
+ *  Name of the property used to get the token's new value.
+ * @param {object} dictionary
+ *  Object of transformed tokens and helper fns provided by style-dictionary.
+ * @returns {object} Token object with an updated value.
+ */
+function transformTokenValue(token, prop, dictionary) {
+  let originalVal = token.original.value[prop];
+  if (dictionary.usesReference(originalVal)) {
+    let refs = dictionary.getReferences(originalVal);
+    return { ...token, value: `var(--${refs[0].name})` };
```

Comment:
I'm not convinced this is the correct return here, but currently it works. If the value here is complex, like a `light-dark()` or `color-mix()` that references multiple vars, then the whole thing is replaced with a single `var(--foo)` statement

ie `"prefersContrast": "light-dark({color.gray.90}, {color.gray.05})"` becomes `var(--color-gray.90)`

We can solve that now if you want, or it can go in your follow up you showed me this afternoon, I think it would be something like this:

```suggestion
    for (let ref of refs) {
      originalVal = originalVal.replaceAll(`{${ref.path.join(".")}}`, `var(--${ref.name})`);
    }
    return { ...token, value: originalVal };
```

Reply:
Yeah this caused some problems in the follow up patch so I already fixed it there in pretty much the same way you're recommending 😄 

<img width="428" alt="Screenshot 2024-01-29 at 10 11 59 AM" src="https://github.com/FirefoxUX/gecko-dev/assets/15138046/91db1d8a-ca40-427c-b6c0-0f4ebd3f1ef5">

Just gonna leave it in the follow up to make my life a little easier...